src/layer/AbstractSentinelHubV1OrV2Layer.ts

```diff
@@ -140,4 +155,53 @@ export class AbstractSentinelHubV1OrV2Layer extends AbstractLayer {
 
     return response.data.map((date: string) => moment.utc(date).toDate());
   }
+
+  public async getStatsAndHistogram(params: GetStatsAndHistogramParams): Promise<StatsAndHistogram> {
+    if (!params.geometry) {
+      throw new Error('Parameter "geometry" needs to be provided');
+    }
+    if (!params.crs) {
+      throw new Error('Parameter "crs" needs to be provided');
+    }
+    if (!params.resolution) {
+      throw new Error('Parameter "resolution" needs to be provided');
+    }
+    if (!params.fromTime || !params.toTime) {
+      throw new Error('Parameter "fromTime" and "toTime" needs to be provided');
+    }
+
+    const payload: FisPayload = {
+      layer: this.layerId,
+      crs: params.crs.authId,
+      geometry: WKT.convert(params.geometry),
+      time: `${moment.utc(params.fromTime).format('YYYY-MM-DDTHH:mm:ss') + 'Z'}/${moment
+        .utc(params.toTime)
+        .format('YYYY-MM-DDTHH:mm:ss') + 'Z'}`,
+      resolution: undefined,
+      bins: params.binAmount || 5,
+      type: HistogramType.EQUALFREQUENCY,
+      ...this.getStatsAndHistogramAdditionalParameters(),
+    };
+    // When using CRS=EPSG:4326 one has to add the "m" suffix to enforce resolution in meters per pixel
+    if (params.crs.authId === CRS_EPSG4326.authId) {
+      payload.resolution = params.resolution + 'm';
+    } else {
+      payload.resolution = params.resolution;
+    }
+    if (params.evalscript) {
+      payload.evalscript = btoa(params.evalscript);
+      payload.evalsource = this.getEvalsource();
+    }
+
+    const { data } = await axios.get(this.dataset.shServiceHostname + 'v1/fis/' + this.instanceId, {
+      params: payload,
+    });
+    // convert date strings to Date objects
+    Object.keys(data).forEach(key => {
+      data[key].forEach((dailyStats: StatsPerChannel) => {
+        dailyStats.date = new Date(dailyStats.date);
+      });
```

Comment:
Maybe it's just me, but with `forEach` I find it surprising that the original array changes. Maybe just a matter of taste, but I would suggest writing this a bit differently:
```javascript
for (let c in data) {
  data[c] = data[c].map((dailyStats: StatsPerChannel) => ({
    ...dailyStats,
    date: new Date(dailyStats.date);
  }));
}
```
(didn't run it)

Reply:
I just think it is a matter of taste, and I am indifferent between the 2. Changed it to use the for loop and map though